packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -150,7 +150,11 @@ function setChecked(
   } else if (isSet(value)) {
     el.checked = value.has(vnode.props!.value)
   } else if (value !== oldValue) {
-    el.checked = looseEqual(value, getCheckboxValue(el, true))
+    if (el._trueValue === undefined && el._falseValue === undefined) {
+      el.checked = looseEqual(!!value, getCheckboxValue(el, true))
+    } else {
+      el.checked = looseEqual(value, getCheckboxValue(el, true))
+    }
```

Comment:
```suggestion
    if ('_trueValue' in el || '_falseValue' in el) {
      el.checked = looseEqual(value, getCheckboxValue(el, true))
    } else {
      el.checked = !!value
    }
```

1. Using `in` rather than `=== undefined` would be more consistent with `getCheckboxValue`, allowing `undefined` to be used as a value. I've flipped the branches to avoid negating everything.
2. `looseEqual(!!value, getCheckboxValue(el, true))` simplifies down to just `!!value` if `trueValue` isn't specified.
3. On the latest `main`, `el.checked` has been replaced with a local variable called `checked`, but otherwise the logic should be the same.